procountor/client.py

```diff
@@ -115,11 +115,29 @@ def _get_token(self):
             'content-type': 'application/x-www-form-urlencoded'
         }
 
-        r = requests.post(self.api_url + 'oauth/token/', params=params, headers=headers)
+        url = self.api_url + 'oauth/token/'
+        response = requests.post(url, params=params, headers=headers)
+        status_code = response.status_code
 
-        self.access_token = r.json()['access_token']
+        if status_code != 200:
```

Comment:
why not use response.raise_for_status() ?

would make this simpler, just add:

```
        response.raise_for_status()
```
after the request. No need to check for the status codes after that.

https://docs.python-requests.org/en/master/user/quickstart/#errors-and-exceptions

Reply:
The idea was given a more special error message and hints like that check api version number. More informative than just 404 default.